components/brave_ads/core/browser/internals/ads_internals_handler.h

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) 2024 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#ifndef BRAVE_COMPONENTS_BRAVE_ADS_CORE_BROWSER_INTERNALS_ADS_INTERNALS_HANDLER_H_
+#define BRAVE_COMPONENTS_BRAVE_ADS_CORE_BROWSER_INTERNALS_ADS_INTERNALS_HANDLER_H_
+
+#include <optional>
+#include <string>
+
+#include "base/memory/weak_ptr.h"
+#include "base/values.h"
+#include "brave/components/brave_ads/core/public/service/ads_service_callback.h"
+#include "brave/components/services/bat_ads/public/interfaces/bat_ads.mojom.h"
+#include "components/prefs/pref_change_registrar.h"
+#include "mojo/public/cpp/bindings/receiver.h"
+#include "mojo/public/cpp/bindings/remote.h"
+
+class PrefService;
+
+namespace brave_ads {
+class AdsService;
+}  // namespace brave_ads
+
+class AdsInternalsHandler : public bat_ads::mojom::AdsInternals {
+ public:
+  AdsInternalsHandler(brave_ads::AdsService* ads_service, PrefService* prefs);
+
+  AdsInternalsHandler(const AdsInternalsHandler&) = delete;
+  AdsInternalsHandler& operator=(const AdsInternalsHandler&) = delete;
+
+  AdsInternalsHandler(AdsInternalsHandler&&) noexcept = delete;
+  AdsInternalsHandler& operator=(AdsInternalsHandler&&) noexcept = delete;
```

Comment:
We should be only deleting the copy constructor/assignment. We don't need to explicitly delete the move assignment according to this guideline, but that's what we are asked to do in this guideline for chromium: https://chromium.googlesource.com/chromium/src/+/main/styleguide/c++/c++-dos-and-donts.md#explicitly-declare-class-copyability_movability

> Declare or delete these operations in the public section, between other constructors and the destructor; DISALLOW_COPY_AND_ASSIGN is deprecated. **For a non-copyable/movable type, delete the copy operations (the move operations will be implicitly deleted)**; otherwise, declare either copy operations, move operations, or both (a non-declared pair will be implicitly deleted). Always declare or delete both construction and assignment, not just one (which can introduce subtle bugs).

I know the wording is confusing because of the previous paragraph, but you can also confirm that that's the practice in most of upstream code.